server/impl/impl.go

```diff
@@ -8,36 +8,53 @@ import (
 	"github.com/gorilla/sessions"
 )
 
+// An Implementation provides all server endpoints for the app.
 type Implementation struct {
 	Options struct {
 		Production bool `long:"production" description:"Run in production mode"`
 	}
 
-	sessionStore     *sessions.CookieStore // lazily initialized; use SessionStore() instead!
+	sessionStore     sessions.Store // could be lazily initialized; use SessionStore() instead!
 	sessionStoreInit sync.Once
 }
 
+// NewImplementation returns a new Implementation intended for production,
+// with a sessions.CookieStore as the internal session store.
 func NewImplementation() *Implementation {
 	return new(Implementation)
 }
 
+// NewImplementation returns a new Implementation with the provided session store.
```

Comment:
I assume this should be NewImplementationWithSessionStore
```suggestion
// NewImplementationWithSessionStore returns a new Implementation with the provided session store.
```